lxd/storage/backend_lxd.go

```diff
@@ -2677,6 +2677,50 @@ func (b *lxdBackend) MountInstance(inst instance.Instance, op *operations.Operat
 	return mountInfo, nil
 }
 
+// DelegateInstance sets up an instance volume for delegation.
+func (b *lxdBackend) DelegateInstance(inst instance.Instance) error {
+	// Delegation only works for running instances.
+	if inst.InitPID() < 1 {
+		return nil
+	}
+
+	// Delegation is only relevant to containers.
+	if inst.Type() != instancetype.Container {
+		return nil
+	}
+
+	l := b.logger.AddContext(logger.Ctx{"project": inst.Project().Name, "instance": inst.Name()})
+	l.Debug("DelegateInstance started")
+	defer l.Debug("DelegateInstance finished")
+
+	// Check we can convert the instance to the volume type needed.
+	volType, err := InstanceTypeToVolumeType(inst.Type())
+	if err != nil {
+		return err
+	}
+
+	contentType := InstanceContentType(inst)
+
+	// Get the volume.
+	var vol drivers.Volume
+	volStorageName := project.Instance(inst.Project().Name, inst.Name())
+
+	// Load storage volume from database.
+	dbVol, err := VolumeDBGet(b, inst.Project().Name, inst.Name(), volType)
+	if err != nil {
+		return err
+	}
+
+	// Generate the effective root device volume for instance.
+	vol = b.GetVolume(volType, contentType, volStorageName, dbVol.Config)
+	err = b.applyInstanceRootDiskOverrides(inst, &vol)
+	if err != nil {
+		return err
+	}
+
+	return b.driver.DelegateVolume(vol, inst.InitPID())
```

Comment:
The return value of `InitPID()` should be stored in a variable as it's called twice.

Reply:
Good catch, will fix.

Reply:
Fixed